File: src/main/java/com/dochiri/hexagonal/presentation/auth/response/LoginUserResponse.java
package com.dochiri.hexagonal.presentation.auth.response;

import com.dochiri.hexagonal.application.user.dto.LoginUserResult;

public record LoginUserResponse(
        String id,
        String email,
        String role,
        String accessToken
) {
    public static LoginUserResponse from(LoginUserResult result) {
        return new LoginUserResponse(
                result.id(),
                result.email(),
                result.role(),
                result.accessToken()
        );
    }
}

File: src/main/kotlin/com/dochiri/hexagonal/presentation/auth/response/LoginUserResponse.kt
package com.dochiri.hexagonal.presentation.auth.response

import com.dochiri.hexagonal.application.user.dto.LoginUserResult

data class LoginUserResponse(
    val id: String,
    val email: String,
    val role: String,
    val accessToken: String
) {
    companion object {
        fun from(result: LoginUserResult): LoginUserResponse =
            LoginUserResponse(
                result.id,
                result.email,
                result.role,
                result.accessToken
            )
    }
}
